test/e2e/specs/onboarding/signup__with-theme-premium.ts

```diff
@@ -0,0 +1,204 @@
+ */
+describe( 'Lifecyle: Premium theme signup, onboard, launch and cancel subscription', function () {
+	const planName = 'Premium';
+	let themeSlug: string | null = null;
+
+	const testUser = DataHelper.getNewTestUser( {
+		usernamePrefix: 'ftmepremium',
+	} );
+
+	let page: Page;
+	let newUserDetails: NewUserResponse;
+	let newSiteDetails: NewSiteResponse;
+
+	beforeAll( async () => {
+		page = await browser.newPage();
+	} );
+
+	describe( 'Signup', function () {
+		let cartCheckoutPage: CartCheckoutPage;
+		let signupPickPlanPage: SignupPickPlanPage;
+
+		beforeAll( async function () {
+			await BrowserManager.setStoreCookie( page, { currency: 'USD' } );
+		} );
+
+		it( 'Navigate to Theme Showcase', async function () {
+			const themesPage = new ThemesPage( page );
+			await themesPage.visitShowcase();
+		} );
+
+		it( 'Selects a Premium theme', async function () {
+			await page.locator( 'div.theme-card:has(div.premium-badge)' ).first().click();
+		} );
+
+		it( 'Navigate to Signup page', async function () {
+			const themeDetailsPage = new ThemesDetailPage( page );
+
+			const pageMatch = new URL( page.url() ).pathname.match( 'theme/(.*)/?' );
+
+			themeSlug = Array.isArray( pageMatch ) ? pageMatch[ 1 ] : null;
+
+			await themeDetailsPage.pickThisDesign();
+		} );
+
+		it( 'Sign up as new user', async function () {
+			const userSignupPage = new UserSignupPage( page );
+			newUserDetails = await userSignupPage.signupSocialFirstWithEmail( testUser.email );
+		} );
+
+		it( 'Skip domain selection', async function () {
+			const domainSearch = new DomainSearchComponent( page );
+
+			await domainSearch.search( testUser.siteName );
+			await domainSearch.selectDomain( `${ testUser.siteName }.wordpress.com` );
+		} );
+
+		it( `Select WordPress.com ${ planName } plan`, async function () {
+			signupPickPlanPage = new SignupPickPlanPage( page );
+			newSiteDetails = await signupPickPlanPage.selectPlan( planName );
+		} );
+
+		it( 'See secure payment', async function () {
+			cartCheckoutPage = new CartCheckoutPage( page );
+			await cartCheckoutPage.validateCartItem( `WordPress.com ${ planName }` );
+		} );
+
+		it( 'Apply coupon', async function () {
+			await cartCheckoutPage.enterCouponCode( SecretsManager.secrets.testCouponCode );
+		} );
+
+		it( 'Enter billing and payment details', async function () {
+			const paymentDetails = DataHelper.getTestPaymentDetails();
+			await cartCheckoutPage.enterBillingDetails( paymentDetails );
+			await cartCheckoutPage.enterPaymentDetails( paymentDetails );
+		} );
+
+		it( 'Make purchase', async function () {
+			await cartCheckoutPage.purchase( { timeout: 90 * 1000 } );
+		} );
+
+		it( 'Skip upsell if present', async function () {
+			const selector = 'button[data-e2e-button="decline"]';
+			await page.waitForSelector( selector );
+
+			const locator = page.locator( selector );
+
+			try {
+				await locator.click( { timeout: 2 * 1000 } );
```

Comment:
Can we _know_ whether the upsell will be present or not?

In general, I'd encourage to avoid this kind of validation because:
1. It will likely still throw the timeout error in the trace file, which can be misleading when inspecting.
2. It can cause the test to be flaky if the timeout isn't sufficient for some reason, and timings in E2E tests can vary depending on the current CI mood. 🌦️

Reply:
Hi @WunderBart, the upsell always appears for Premium plans, which what we always select for Premium themes in this test.

Maybe I should rename the test to "Skip upsell to Business plan" to be less confusing.